src/plugins/intel_gpu/src/kernel_selector/cl_kernels/mvn_gpu_bfyx_opt.cl

```diff
@@ -4,45 +4,57 @@
 
 #include "include/batch_headers/fetch_data.cl"
 
+#if !IS_DYNAMIC
 __attribute__((reqd_work_group_size(LWS, 1, 1)))
+#endif
 KERNEL (mvn_gpu_bfyx_opt)(
+    OPTIONAL_SHAPE_INFO_ARG
     const __global INPUT0_TYPE* input,
     __global OUTPUT_TYPE* restrict output
 #if HAS_FUSED_OPS_DECLS
     , FUSED_OPS_DECLS
 #endif
 ) {
-    const uint data_set_idx = get_global_id(1);     //in processing of which data set this WI participates?
-    const uint workers_per_data_set = LWS;          //how many WI participates in processing of one data set
-    const uint in_data_set_idx = get_global_id(0);  //this WI's id in group of items processing single data set
-    const uint data_set_size = DATA_SET_SIZE;       //how many elements are in one data set
-    const uint data_sets_count = DATA_SETS_COUNT;   //how many data sets are in the processing payload
+    const uint data_set_idx = get_global_id(1);                   // in processing of which data set this WI participates?
+    const uint workers_per_data_set = get_local_size(0);          // how many WI participates in processing of one data set
+    const uint in_data_set_idx = get_global_id(0);                // this WI's id in group of items processing single data set
+    const uint data_set_size = DATA_SET_SIZE;                     // how many elements are in one data set
+    const uint data_sets_count = DATA_SETS_COUNT;                 // how many data sets are in the processing payload
+#if IS_DYNAMIC
+    // since workers_per_data_set is calculated by power of 2
+    // items_num can be calculated by dividing data_set_size by power of 2
+    const uint items_num = data_set_size / workers_per_data_set;  // how many elements are processed per one WI
+    const uint leftovers = data_set_size % workers_per_data_set;
+#else // IS_DYNAMIC
+    const uint items_num = ITEMS_NUM;
+    const uint leftovers = LEFTOVERS;
+#endif // IS_DYNAMIC
 
     const uint data_set_offset = data_set_idx * data_set_size;
     const uint my_data_offset = data_set_offset + in_data_set_idx;
 
     float my_sum = 0;
     float tmp;
 
-    __local float lg_storage[LWS];
+    __local float lg_storage[SLM_SIZE];
```

Comment:
What do you think of replacing `SLM_SIZE` with `workers_per_data_set`?
```suggestion
    __local float lg_storage[workers_per_data_set];
```

Reply:
It is not constant but can be variable in dynamic shape, so  it will be compiled at build time for shape agnostic kernel.